default/python/common/configure_logging.py

```diff
@@ -128,19 +128,27 @@ def __init__(self, level):
 
     def write(self, msg):
         # Grab the caller's call frame info
-        stack = inspect.stack()
-        if len(stack) > 1:
-            (_, filename, line_number, function_name, _, _) = inspect.stack()[1]
+        if hasattr(sys, '_getframe'):
+            frame = sys._getframe(1)
+            try:
+                line_number = frame.f_lineno
+                function_name = frame.f_code.co_name
+                filename = frame.f_code.co_filename
+                if string.lower(filename[-4:]) in ('.pyc', '.pyo'):
+                    filename = filename[:-4] + '.py'
```

Comment:
I would remove these two lines changing the suffix of the compiled/optimized byte code to indicate that it is running from the text file.  It is not true.  It makes it more difficult for people to determine if they are running from stale compiled (.pyc) files if the log incorrectly indicates that they are running the current python (.py) files.